### ISSUE TITLE
protein-translation: FromCodon tests force less idiomatic solutions.

### ISSUE THREAD

**Issue by hilary** [exercism/go#1132]
Please bear with me while I explain, as I'm still quite new to golang.

When solving protein-translation, my inclination was to use an error to signal that `FromCodon` had hit a stop codon (in the same way that `io` uses an error to signal `EOF`. If there were no tests for `FromCodon`, I could do that (or not, depending on my understanding of the language). But because there are tests for `FromCodon` that don’t collect (or ignore) error responses, I’m forced to treat `STOP` like any other response, which produces what feels like far less idiomatic code. (The effect cascades into how I write `FromRNA` as well.)

I asked about this on slack and @kytrinyx said "It would also provides an opportunity to implement something in the style of `os.IsNotExist(err)`." I [looked up the function](https://golang.org/pkg/os/#IsNotExist). Yup! Exactly! I was all set to stretch my wings and fly and ... grounded 😝

**Comment by sebito91** [comment]
I think this is a cool idea! I think the best way around this is to just call out to a helper function of the same name. Usually when I'm creating packages I'll have a publicly exported version of the function, in this case `FromCodon` but just return the value from my helper version that obfustcates the actual content (just like your example doc ref -- https://golang.org/src/os/error.go?s=2154:2185#L65).

In this case, you can create the custom err value yourself and then just do something like this...

```
type customError error
var MyCustomError customError = errors.New("custom error")

func FromCodon(input string) string {
    str, err := fromCodon(input); 
    if err != nil && isMyErr(err) {
        // do something here like return empty or STOP
    }
    return str
}

func fromCodon(input string) (string, error) {
    // do your stuff here and return string, error at the end
}

func isMyErr(err error) bool {
    switch err.(type) {
    case customError:
        return true
    }
    return false
}
```

**Comment by sebito91** [comment]
And I could be completely incorrect too :D

**Comment by hilary** [comment]
It's a cool workaround, but a better solution would be to fix the implementation of the problem in the track.

Hey, golang track maintainers, I'm gonna take a pass at doing just that...

**Comment by hilary** [comment]
Fixed by #1134
